### PR TITLE
test(*): fix Windows tests

### PR DESCRIPTION
**Pull request by rodcloutier**
This PR intent is to augment support for running on Windows.

The current assumption is that all paths will use the posix path separator "/". Windows uses the "\" path separator. 

I propose that we sanitize the paths and avoid using filePath.Join when composing path like values. 

* Search returned chart names with "\". Fixed to always return names with "/" to meet the expectations of the product.
* Fixed all the test to run on Windows

### CODE REVIEW

File: pkg/tiller/release_server.go [hunk 1]
```diff
@@ -735,7 +734,8 @@ func (s *ReleaseServer) renderResources(ch *chart.Chart, values chartutil.Values
 	for k, v := range files {
 		if strings.HasSuffix(k, notesFileSuffix) {
 			// Only apply the notes if it belongs to the parent chart
-			if k == filepath.Join(ch.Metadata.Name, "templates", notesFileSuffix) {
+			// Note: Do not use filePath.Join since it creates a path with \ which is not expected
+			if k == strings.Join([]string{ch.Metadata.Name, "templates", notesFileSuffix}, "/") {
```
Comment (technosophos) [inline pkg/tiller/release_server.go]: I think `path.Join` might do what we want, since I believe it always uses `/`.

**Comment by technosophos** [comment]
Wow, this is fabulous! Thanks. I left one comment about the `path.Join` thing, and otherwise this all looks great to me. @mattfarina had suggested a way we could run automated tests on Windows to match the CircleCI. This would certainly get us closer. #1606

**Comment by rodcloutier** [comment]
@technosophos Will updating the fix for path.Join, I ran into a problem with the plugins tests. Currently the plugin are all written using the assumption that bash script can be run directly on the OS. Unfortunately, on Windows this is not the case. I was thinking to skip the test for the time being and open a issue to support this (we could detect that the script is a bash script and prefix the command with bash to make it run on Windows as well). Any thoughts?

**Comment by technosophos** [comment]
@rodcloutier I am going to merge this as-is, then. Let me know what you think about #1638, and if that will help make plugins better I will work on that.

**Pull request merged**